src/auth/ha-authorize.ts

```diff
@@ -1,251 +1,516 @@
+  color: var(--error-color);
+}
+svg.info {
+  color: var(--tinted-text-color);
+}
+p {
+  margin: 0;
+}
+p.secondary {
+  opacity: 0.8;
+}
+form {
+  display: contents;
+}
+button {
+  font-family: inherit;
+  font-weight: 500;
+  border: none;
+  cursor: pointer;
+  white-space: nowrap;
+}
 
-      if (authProviders.length === 0) {
-        this._error = "No auth providers returned. Unable to finish login.";
-        return;
-      }
 
-      this._authProviders = authProviders;
-      this._authProvider = authProviders[0];
-    } catch (err: any) {
-      this._error = "Unable to fetch auth providers.";
-      // eslint-disable-next-line
-      console.error("Error loading auth providers", err);
-    }
+div.error {
+  display: flex;
+  gap: 8px;
+  justify-content: center;
+  align-items: center;
+  padding: 12px 14px;
+  border-radius: 16px;
+  background-color: rgb(var(--rgb-error-color), 0.2);
+  
+  align-self: stretch;
+  margin-bottom: 8px;
+}
+
+ha-authorize-field {
+  display: flex;
+  position: relative;
+  overflow: hidden;
+  flex-direction: column;
+
+  align-self: stretch;
+  border-radius: 4px 4px 0 0;
+}
+input:is([type=text], [type=number], [type=password]), select {
+  background-color: transparent;
+  color: inherit;
+  border: none;
+  height: 48px;
+  padding: 0 16px;
+  font-family: inherit;
+  font-size: inherit;
+}
+input::placeholder {
+  color: inherit;
+  opacity: var(--text-opacity);
+}
+input::-webkit-outer-spin-button,
+input::-webkit-inner-spin-button {
+  appearance: none;
+}
+input[type=number] {
+  appearance: textfield;
+}
+input:focus-visible + .bar, select:focus-visible + .bar {
+  background-color: var(--primary-color);
+  height: 4px;
+}
+.bar {
+  position: absolute;
+  background-color: var(--bar-color);
+  height: 2px;
+  width: 100%;
+  bottom: 0;
+  pointer-events: none;
+  transition: all 200ms;
+}
+ha-authorize-field > button {
+  position: absolute;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  opacity: var(--text-opacity);
+
+  width: 40px;
+  height: 40px;
+  top: 4px;
+  right: 4px;
+  padding: 0;
+  border-radius: 20px;
+  background-color: transparent;
+}
+ha-authorize-field > button:hover {
+  background-color: inherit;
+}
+input:is([type=text], [type=number], [type=password]):focus-visible, select:focus-visible {
+  outline: none;
+}
+
+select {
+  padding: 20px 12px 6px 12px;
+}
+.name {
+  position: absolute;
+  left: 16px;
+  top: 6px;
+  font-size: 12px;
+  opacity: var(--text-opacity);
+  pointer-events: none;
+}
+@supports (-moz-appearance: none) {
+  select {
+    padding-left: 16px;
+    padding-right: 16px;
   }
+}
+
+label {
+  display: flex;
+  align-items: center;
+  height: 48px;
+  align-self: stretch;
+  cursor: pointer;
+}
+input[type=checkbox] {
+  width: 18px;
+  height: 18px;
+  margin: 0 16px 0 0;
+  cursor: pointer;
+}
 
-  private async _handleAuthProviderPick(ev) {
-    this._authProvider = ev.detail;
+ha-authorize-field + .buttons {
```

Comment:
Keep the styling of elements with those elements instead of here and the template

Reply:
> with those elements

What would this look like? Something like a const at the top level and a `<style>${styles}</style>` at each render return, or did you mean something else? (Also you could argue having all the styles in one place makes more sense when they aren't scoped)

> and the template

Most of the template styling is there to prevent a layout shift, so everything except the `ha-authorize-field` styling isn't going anywhere.